src/main/java/org/opensearch/ad/task/ADTaskManager.java

```diff
@@ -1482,11 +1484,13 @@ private void createNewADTask(
         AnomalyDetector detector,
         DetectionDateRange detectionDateRange,
         User user,
+        String realtimeADCoordinatingNode,
         ActionListener<AnomalyDetectorJobResponse> listener
     ) {
         String userName = user == null ? null : user.getName();
         Instant now = Instant.now();
         String taskType = getADTaskType(detector, detectionDateRange).name();
+        String coordinatingNode = detectionDateRange == null ? realtimeADCoordinatingNode : clusterService.localNode().getId();
```

Comment:
Check `ADTaskManager#startHistoricalAnalysis` line 407, it will choose owning node and forward START action to coordinating node. Then check `ForwardADTaskTransportAction` line 125, it starts historical analysis on coordinating node.

Reply:
I see.  Is it more clearer that you repurpose realtimeADCoordinatingNode as coordinatingNode and let the caller decide what node to pass in? Currently, the precondition is that the caller of this method is from a coordinating node.  It is easy for others to forget and make mistakes.

Reply:
Sure, will tune this part.